app/invites/server/functions/useInviteToken.js

```diff
@@ -0,0 +1,38 @@
+import { Meteor } from 'meteor/meteor';
+
+import { Invites, Rooms, Users } from '../../../models';
+import { validateInvite } from './validateInviteToken';
+import { addUserToRoom } from '../../../lib/server/functions/addUserToRoom';
+
+export const useInviteToken = (userId, token) => {
+	if (!userId) {
+		throw new Meteor.Error('error-invalid-user', 'The user is invalid', { method: 'useInviteToken', field: 'userId' });
+	}
+
+	if (!token) {
+		throw new Meteor.Error('error-invalid-token', 'The invite token is invalid.', { method: 'useInviteToken', field: 'token' });
+	}
+
+	const inviteData = Invites.findOneById(token);
+	const room = inviteData && Rooms.findOneById(inviteData.rid, { fields: { _id: 1, fname: 1, t: 1 } });
+
+	if (!validateInvite(inviteData, room)) {
```

Comment:
validateInviteToken could be used here and return the invite and room to be used later.